.github/actions/build-and-publish-pr-preview/action.yml

```
@@ -46,7 +53,7 @@ runs:
         du --max-depth=2 -h
     - name: Publish preview
       uses: afc163/surge-preview@v1
-      if: steps.surge-preview-tools.outputs.can-run-surge-command == 'true'
+      if: steps.surge-preview-tools.outputs.can-run-surge-command == 'true' && inputs.publish-preview == 'true'
```

inputs.publish-preview doesn't exist
```suggestion
      if: steps.surge-preview-tools.outputs.can-run-surge-command == 'true'
```